scripts/util.py

```diff
@@ -701,6 +701,42 @@ def cached_property(f):
     return property(lru_cache()(f))
 
 
+def retry(max_retries: int, init_wait_time: float, warn_msg: str, exc_type: Exception):
+    """Retries functions that raises the exception exc_type.
+    Retry time is increased by a factor of two for every iteration.
+
+    Args:
+        max_retries (int): Maximum number of retries
+        init_wait_time (float): Initial wait time in secs
+        warn_msg (str): Message to print during retries
+        exc_type (Exception): Exception type to check for
+    """
+
+    def decorator(f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            retry = 0
+            secs = init_wait_time
+            captured_exc = None
+            if retry < max_retries:
+                try:
+                    return f(*args, **kwargs)
+                except exc_type as e:
+                    captured_exc = e
+                    log.warn(f"{warn_msg}, retrying in {secs}")
+                    sleep(secs)
+                    retry += 1
+                    secs *= 2
+            elif captured_exc is not None:
+                raise captured_exc
+            else:
+                raise exc_type()
```

Comment:
The motivation for this `else` block is not clear. AFAIK It's only possibly if `max_retries == 0` , that is IMHO an invalid argument for `retry` and should be handled during construction of the decorator.

Reply:
Added error handling, removed else block.